File: src/main/java/org/processmining/models/GenerationDescription.java
package org.processmining.models;

/**
 * Created by Ivan Shugurov on 30.10.2014.
 */
public interface GenerationDescription
{
    int getNumberOfLogs();

    void setNumberOfLogs(int numberOfLogs);

    int getNumberOfTraces();

    void setNumberOfTraces(int numberOfTraces);

    int getMaxNumberOfSteps();

    void setMaxNumberOfSteps(int maxNumberOfSteps);

    boolean isUsingTime();

    boolean isUsingResources();

    boolean isUsingLifecycle();

    boolean isRemovingEmptyTraces();

    boolean isRemovingUnfinishedTraces();
}

File: src/main/java/org/processmining/models/GenerationDescription.kt
package org.processmining.models

/**
 * Created by Ivan Shugurov on 30.10.2014.
 */
interface GenerationDescription {
    var numberOfLogs: Int
    
    var numberOfTraces: Int
    
    var maxNumberOfSteps: Int
    
    val isUsingTime: Boolean
    
    val isUsingResources: Boolean
    
    val isUsingLifecycle: Boolean
    
    val isRemovingEmptyTraces: Boolean
    
    val isRemovingUnfinishedTraces: Boolean
}
